### PR TITLE
New Tutorial: Easily Consume AWS Secrets Manager Secrets from your Amazon EKS Workloads

### PR DESCRIPTION
**Pull request by rstebaws**
# Content Pull Request Checklist

Please check to make sure your content submission meets the following requirements by adding an `x` between the brackets:

- [x] The title of this pull request reflects the title of my content.
- [x] I have reviewed the [Content Review Checklist](https://github.com/build-on-aws/content/blob/main/CONTENT_REVIEW_CHECKLIST.md).
- [x] This pull request contains one piece of content (post or page).

**Important: if you are submitting a PR to update existing content, please ensure you [sync'ed your fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) first before you created your branch.**

Is this pull request linked to a Content Proposal or Idea Suggestion Issue? If so, please link to it below using the number or full link.

```text
Resolves #5

or

Content for https://github.com/build-on-aws/content/issues/5
```

**By submitting this pull request, I confirm I own this content, have permission to use each image included in this content, and that it will be released under the [CC BY-SA 4.0 SA License](/LICENSE).**

### CODE REVIEW

**Review by scribwell — APPROVED** [review]

**Comment by scribwell** [comment]
@DalM2023 This tutorial is ready for your review.

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+  restartPolicy: Always
+```
+
+Apply the YAML manifest. 
+
+```bash
+`kubectl apply ``-``f eksdemo-app.yaml`
+```
+
+To verify the pod was created successfully, run the following command.
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:368]:
```suggestion
To verify the pod was created successfully, run the following command:
```

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+```bash
+kubectl get pod busybox
+```
+
+You should see the following output. Make sure the pod  `STATUS` is `Running`.
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:374]:
```suggestion
You should see the following output. Make sure the pod  `STATUS` is `Running`:
```

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+helm uninstall -n kube-system csi-secrets-store
+helm uninstall -n kube-system secrets-provider-aws
+
+# Delete IAM Policy
+aws iam delete-policy --policy-arn $POLICY_ARN
+```
+
+## Conclusion
+
+Upon completion of this tutorial, you will have successfully set up an integration between AWS Secrets Manager and your Amazon EKS cluster. This integration allows you to centralize the management of your application secrets, while easily consuming these secrets from your workloads running on EKS, without complex code modifications. Security and governance of your secrets as well as portability of your applications is improved with minimal overhead. This example can easily be replicated for the various types of secrets your workloads may require such as database credentials, API keys, and more.
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:422]: Please add a sentence with links to other blogs apropos of your topic. We want to invite the reader to stay in the blog repository. For ex:

To learn more about X, visit (LINK) and (LINK).

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 4]
```diff
@@ -0,0 +1,422 @@
+---
+title: "Easily Consume AWS Secrets Manager Secrets from your Amazon EKS Workloads"
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:2]:
```suggestion
title: "Easily Consume AWS Secrets Manager Secrets From Your Amazon EKS Workloads"
```

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+description: "How to integrate AWS Secrets Manager with your EKS workloads to safely store and consume application secrets with minimal code modifications."
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:3]: We aim to keep the descriptions to some 60 characters. Also, they don't need to repeat the same words or messaging that's in the title. It should focus on what problem it solves for the reader. Consider rewriting. EX:  
"Manage app secrets without complex code modifications." 

```suggestion
description: "How to integrate AWS Secrets Manager with your EKS workloads to safely store and consume application secrets with minimal code modifications."
```

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 6]
```diff
@@ -0,0 +1,422 @@
...
+tags:
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:4]: We can only use five tags. Please choose the ones you think are strongest for the piece.

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 7]
```diff
@@ -0,0 +1,422 @@
+---
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:1]: Hi there! 

You have a very well-written blog! It's well-structured, direct and preps reader well for the project. Please see notes throughout the piece for my edits. 

Well done! 👍

**Review by DalM2023 — COMMENTED** [review]
First round edit done. Ready for review.

**Comment by rstebaws** [comment]
@DalM2023 - I have updated with the requested changes. Thanks for the feedback!

**Comment by scribwell** [comment]
@DalM2023 This tutorial is ready for your second pass.

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+secrets-provider-aws kube-system [1 2023-08-07 15](tel:12023080715):39:55.851595668 +0000 UTC deployed secrets-store-csi-driver-provider-aws-0.3.4
+```
+
+## Step 6: Create ASCP SecretProviderClass Resource
+
+In this section, you’re defining the `SecretProviderClass` Kubernetes object, which sets the stage for seamless secrets management within your Kubernetes workloads. This resource acts as a set of instructions for the AWS Secrets and Configuration Provider (ASCP), specifying which secrets to fetch from AWS Secrets Manager and how to mount them into your pods. Note that the SecretProviderClass must be deployed in the same namespace as the workload that references it. To learn more, see [SecretProviderClass documentation](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_csi_driver_SecretProviderClass.html?sc_channel=el&sc_campaign=appswave&sc_content=eks-integrate-secrets-manager&sc_geo=mult&sc_country=mult&sc_outcome=acq).
+
+Create a Kubernetes manifest called `eksdemo-spc.yaml` and paste the following contents into it.
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:275]:
```suggestion
Create a Kubernetes manifest called `eksdemo-spc.yaml` and paste the following contents into it:
```

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+
+  Provider:  aws
+Events:      <none>
+```
+
+## Step 7: Deploy Sample Workload to Consume Secret
+
+In this section, you'll deploy a sample workload to bridge your application and AWS Secrets Manager. By mounting the secret as a file on the workload's filesystem, you'll complete the end-to-end process of securely managing and accessing secrets within your Kubernetes environment. In the pod template you will specify the Secrets Store CSI as the volume driver and then a path to mount your secret, just like you would a traditional volume mount. In this example we will mount the secret in the `/mnt/secrets-store` location.
+
+Create a Kubernetes manifest called `eksdemo-app.yaml` and paste the following contents into it.
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:330]:
```suggestion
Create a Kubernetes manifest called `eksdemo-app.yaml` and paste the following contents into it:
```

File: tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+helm uninstall -n kube-system csi-secrets-store
+helm uninstall -n kube-system secrets-provider-aws
+
+# Delete IAM Policy
+aws iam delete-policy --policy-arn $POLICY_ARN
+```
+
+## Conclusion
+
+Upon completion of this tutorial, you will have successfully set up an integration between AWS Secrets Manager and your Amazon EKS cluster. This integration allows you to centralize the management of your application secrets, while easily consuming these secrets from your workloads running on EKS, without complex code modifications. Security and governance of your secrets as well as portability of your applications is improved with minimal overhead. This example can easily be replicated for the various types of secrets your workloads may require such as database credentials, API keys, and more. To learn more about setting up and managing Amazon EKS for your workloads, check out [Navigating Amazon EKS](https://community.aws/tutorials/navigating-amazon-eks#list-of-all-tutorials?sc_channel=el&sc_campaign=appswave&sc_content=eks-integrate-secrets-manager&sc_geo=mult&sc_country=mult&sc_outcome=acq).
```
Comment (DalM2023) [inline tutorials/navigating-amazon-eks/eks-integrate-secrets-manager/index.md:421]:
```suggestion
Upon completion of this tutorial, you will have successfully set up an integration between AWS Secrets Manager and your Amazon EKS cluster. This integration allows you to centralize the management of your application secrets, while easily consuming these secrets from your workloads running on EKS, without complex code modifications. Security and governance of your secrets as well as portability of your applications is improved with minimal overhead. This example can easily be replicated for the various types of secrets your workloads may require such as database credentials, API keys, and more. 

To learn more about setting up and managing Amazon EKS for your workloads, check out [Navigating Amazon EKS](https://community.aws/tutorials/navigating-amazon-eks#list-of-all-tutorials?sc_channel=el&sc_campaign=appswave&sc_content=eks-integrate-secrets-manager&sc_geo=mult&sc_country=mult&sc_outcome=acq).
```

**Review by DalM2023 — COMMENTED** [review]
2nd round edit done.

**Comment by rstebaws** [comment]
@DalM2023 @scribwell - Changes from second round of review have been committed. Thanks!

**Comment by scribwell** [comment]
@rstebaws Thanks for the quick turn-around! @damonk10 Please review Ryan's revisions as part of your final review. The goal is to publish this article on October 30th.

**Review by damonk10 — APPROVED** [review]

**Pull request merged**